### DISCUSSION TITLE
Help with command using Ubuntu

### DISCUSSION THREAD

**Discussion by GangstaPichu** [discord-tickets/bot#421] [Support (Q&A)] [answered]
I'm trying to run "npm i --production && npm start
" but I don't have npm to work with. I'm not sure if there's anyway around this. I saw that there's things like yarn and what not, but I'm not sure how to even use those properly to make it do what that command is doing. I know it's installing dependencies and such, but I lack enough knowledge to find a good alternative it seems. I've had this issue with npm before. It's like the one command I always have an issue with when guides use it. So, I'm lost on what to do and I'd really love to set the bot up, but I can't progress past this stage. I've tried to find something on my own, but like I said. I just lack the experience and know how I need to pull this off. So, any help is much appreciated!

**Comment by eartharoid** [comment, marked as answer]
NPM is installed alongside Node.js by default, and the easiest way to correctly install Node.js on Ubuntu is to first [install NVM](https://github.com/nvm-sh/nvm#installing-and-updating), then (after restarting your terminal) run `nvm install node`.
You should then be able to do `node -v` and `npm -v` successfully.

**Reply by GangstaPichu** [reply]
I just did that. Am I supposed to get this when doing those commands at the end? 

"node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by node)"

**Reply by eartharoid** [reply]
No, that's not supposed to happen 😄. If for whatever reason you can't get your environment working correctly, you could try running the bot in a Docker container instead.
I can't help you do that with v3, but the v4 beta should work easily with Docker.

**Reply by GangstaPichu** [reply]
Ok, nevermind that. I found something that fixed that? This is what I did just incase I did something wrong here.

sudo apt-get remove nodejs
nvm install 16.15.1

Now when I run the command stuff it's all fine. So, I tried to do the command to setup the bot and I get this:

npm WARN config production Use `--omit=dev` instead.
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /home/ubuntu/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/home/ubuntu/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2023-05-14T21_44_40_989Z-debug-0.log

**Reply by eartharoid** [reply]
It looks like you're running the command in your home directory. You need to `cd` into the directory you downloaded the code into, `bot` by default if you cloned the repository with git.

**Reply by GangstaPichu** [reply]
Oh! Thanks. Lol

I guess I overlooked that. My bad. Thanks! Seems it all works now. Just had those 2 little hiccups. I put what I did down in my notepad for whenever I need it. Thanks a lot for the help! :)
